static/home/js/datamapper.js

```diff
@@ -0,0 +1,3548 @@
+            if (data_types_list[col] === 'Discrete') {
+                Data_styling[col].Datatype = 'Discrete';
+            } else if (data_types_list[col] === 'Continuous') {
+                Data_styling[col].Datatype = 'Continuous';
+            }
+        }
+    });
+
+
+    // Run through data and set min and max values
+
+    function updateDataMinMax(column, key, currentObj) {
+        if (Data_styling[column].Data === "Yes" && Data_styling[column].Datatype === 'Continuous' && currentObj.hasOwnProperty(key)) {
+            if (currentObj[key] > Data_styling[column].Data_max) {
+                Data_styling[column].Data_max = currentObj[key];
+            }
+            if (currentObj[key] < Data_styling[column].Data_min) {
+                Data_styling[column].Data_min = currentObj[key];
+            }
+        }
+    }
+
+    Object.keys(list_data_wow).forEach(function(key) {
+        var currentObj = list_data_wow[key];
+
+        updateDataMinMax('Col1', 'Value2', currentObj);
+        updateDataMinMax('Col2', 'Value4', currentObj);
+        updateDataMinMax('Col3', 'Value6', currentObj);
+        updateDataMinMax('Col4', 'Value8', currentObj);
+    });
+
+    return Data_styling;
+}
+
+// Function to modify the data
+function modifyData(data) {
+    // 1. Change "Other GPCRs" to "Classless" (Layer1)
+    if (data["Other GPCRs"]) {
+      data["Classless"] = data["Other GPCRs"];
+      delete data["Other GPCRs"];
+    }
+  
+    // 2. Change "Other GPCR orphans" to "Classless orphans" (Layer3)
+    Object.keys(data).forEach(layer1Key => {
+      const layer2Data = data[layer1Key];
+      Object.keys(layer2Data).forEach(layer2Key => {
+        const layer3Data = layer2Data[layer2Key];
+        if (layer3Data["Other GPCR orphans"]) {
+          layer3Data["Classless orphans"] = layer3Data["Other GPCR orphans"];
+          delete layer3Data["Other GPCR orphans"];
+        }
+      });
+    });
+  
+    // 3. Remove "Olfactory receptors" from Layer2
+    Object.keys(data).forEach(layer1Key => {
+      const layer2Data = data[layer1Key];
+      if (layer2Data["Olfactory receptors"]) {
+        delete layer2Data["Olfactory receptors"];
+      }
+    });
+  
+    return data;  // Return the modified data
+  }
+
+// ### Visualization Function ###
+function renderDataVisualization(data, location,styling_option,Layout_dict,data_styling,label_conversion_dict,label_names) {
+
+    // ######################
+    // ## Initializzation  ##
+    // ######################
+
+    // Check the visibility of each layer
+    const Layer1_isChecked = d3.select(`#toggle-layer-1`).property('checked');
+    const Layer2_isChecked = d3.select(`#toggle-layer-2`).property('checked');
+    const Layer3_isChecked = d3.select(`#toggle-layer-3`).property('checked');
+
+    let Checklist = [Layer1_isChecked, Layer2_isChecked, Layer3_isChecked];
+
+    // Indentation
+
+    // Data sorter depending on layers being shown:
```

Comment:
couple of commented functions below, will we need these?

Reply:
Not sure, but I think these are still WIP